include/fiction/algorithms/physical_design/planar_layout_from_network_embedding.hpp

```diff
@@ -0,0 +1,1184 @@
+               {{{1, 0}, {1, 0}, {1, 0}, {1, 0}}}   // gap 2
+           }}}},
+         {{// Type F1
+           {{
+               // East
+               {{{0, 0}, {0, 1}, {0, 2}, {0, 1}}},  // gap 0
+               {{{0, 0}, {0, 0}, {0, 1}, {0, 0}}},  // gap 1
+               {{{0, 0}, {0, 0}, {0, 0}, {0, 0}}}   // gap 2
+           }},
+           {{
+               // South
+               {{{3, 0}, {3, 1}, {0, 0}, {0, 0}}},  // gap 0
+               {{{3, 0}, {3, 1}, {0, 0}, {0, 0}}},  // gap 1
+               {{{3, 0}, {3, 1}, {0, 0}, {0, 0}}}   // gap 2
+           }},
+           {{
+               // Free
+               {{{0, 0}, {3, 0}, {3, 1}, {3, 0}}},  // gap 0
+               {{{0, 0}, {0, 0}, {3, 0}, {0, 0}}},  // gap 1
+               {{{0, 0}, {0, 0}, {0, 0}, {0, 0}}}   // gap 2
+           }}}},
+         {{// Type F2
+           {{
+               // East
+               {{{1, 0}, {1, 0}, {1, 1}, {1, 1}}},  // gap 0
+               {{{1, 0}, {1, 0}, {1, 0}, {1, 0}}},  // gap 1
+               {{{1, 0}, {1, 0}, {1, 0}, {1, 0}}}   // gap 2
+           }},
+           {{
+               // South
+               {{{2, 0}, {2, 0}, {0, 0}, {0, 0}}},  // gap 0
+               {{{2, 0}, {2, 0}, {0, 0}, {0, 0}}},  // gap 1
+               {{{2, 0}, {2, 0}, {0, 0}, {0, 0}}}   // gap 2
+           }},
+           {{
+               // Free
+               {{{1, 0}, {1, 0}, {2, 0}, {2, 0}}},  // gap 0
+               {{{1, 0}, {1, 0}, {1, 0}, {1, 0}}},  // gap 1
+               {{{1, 0}, {1, 0}, {1, 0}, {1, 0}}}   // gap 2
+           }}}},
+         {{// Type F3
+           {{
+               // East
+               {{{0, 0}, {0, 0}, {0, 0}, {0, 0}}},  // gap 0
+               {{{0, 0}, {0, 0}, {0, 0}, {0, 0}}},  // gap 1
+               {{{0, 0}, {0, 0}, {0, 0}, {0, 0}}}   // gap 2
+           }},
+           {{
+               // South
+               {{{3, 0}, {3, 0}, {0, 0}, {0, 0}}},  // gap 0
+               {{{3, 0}, {3, 0}, {0, 0}, {0, 0}}},  // gap 1
+               {{{3, 0}, {3, 0}, {0, 0}, {0, 0}}}   // gap 2
+           }},
+           {{
+               // Free
+               {{{0, 0}, {0, 0}, {3, 0}, {3, 0}}},  // gap 0
+               {{{0, 0}, {0, 0}, {0, 0}, {0, 0}}},  // gap 1
+               {{{0, 0}, {0, 0}, {0, 0}, {0, 0}}}   // gap 2
+           }}}}}};
+    return array;
+}
+
+/**
+ * Computes the fan-out connection type of a node based on its successors' fan-in structures.
+ * Assumes exactly two fan-outs, which are ordered by rank position.
+ *
+ * Returns:
+ *  - 0: Both fan-outs have a single fan-in.
+ *  - 1: First fan-out has one fan-in, second has multiple.
+ *  - 2: Second fan-out has one fan-in, first has multiple.
+ *  - 3: Both fan-outs have multiple fan-ins.
+ *
+ * @tparam Ntk Logic network type.
+ * @param ntk  Logic network containing the node.
+ * @param n    Node for which to determine the fan-out connection type.
+ * @return     Integer code (0–3) indicating the fan-out connection pattern.
```

Comment:
An `enum class` type might be more appropriate here. And probably, `uint8_t` is a more appropriate representation since so few status codes are actually in use.